crates/type-system/src/ontology/uri/error.rs

```diff
@@ -19,6 +20,8 @@ impl fmt::Display for ParseBaseUriError {
     }
 }
 
+impl Error for ParseBaseUriError {}
```

Comment:
Can you swap these to `thiserror` impls so they're consistent with other errors like `crates/type-system/src/ontology/entity_type/error.rs:14` and so that we have better format strings. (The current one is obviously not ideal 🙃)

Reply:
I replaced the `Display` implementation with `#[derive(Error])`, I used error messages according to the [API guidelines](https://rust-lang.github.io/api-guidelines/interoperability.html#error-types-are-meaningful-and-well-behaved-c-good-err). 298ed7710c0f31fa83e9360ab56233f3d37974c0

`ParseIntError` does not implement `Serialize`, so I converted it into a `String` 😞

Reply:
Hm, I wonder if we should get rid of the Serialize bound. Anyway looks good as is, thank you